src/util/tubemap.js

```diff
@@ -874,20 +878,35 @@ function setOccupiedUntil(map, read, pathIndex, y, node) {
 
 // compare read segments which are outside of nodes
 // by the y-coord of where they are coming from
-function compareNoNodeReadsByPreviousY(a, b) {
-  const segmentA = reads[a.readIndex].path[a.pathIndex];
-  const segmentB = reads[b.readIndex].path[b.pathIndex];
-  if (segmentA.order === segmentB.order) {
-    // We want to sort in reverse order when the segment is along the reverse-going part of a cycle.
-    // This ensures a loop that starts on the outside, stays on the outside,
-    // and rolls up in order with other loops.
-    if (segmentA?.betweenCycleReverseTraversal && segmentB?.betweenCycleReverseTraversal) {
-      return b.previousY - a.previousY;
-    } else {
-      return a.previousY - b.previousY;
+function compareNoNodeReads(a, b) {
+  // Sort by order by segments
+  const segmentA = readA.path[a.pathIndex];
+  const segmentB = readB.path[b.pathIndex];
+  if (segmentA.order !== segmentB.order) {
+    return segmentA.order - segmentB.order;
+  } 
+  // Sort by reads' source track
+  const readA = reads[a.readIndex];
+  const readB = reads[b.readIndex];
+  if (readA.sourceTrackID !== readB.sourceTrackID){
+    return readA.sourceTrackID - readB.sourceTrackID;
+  }
```

Comment:
The tests are failing because the reads are not defined early enough.

```suggestion
function compareNoNodeReads(a, b) {
  // Get the reads for the segments
  const readA = reads[a.readIndex];
  const readB = reads[b.readIndex];
  // Sort by order by segments
  const segmentA = readA.path[a.pathIndex];
  const segmentB = readB.path[b.pathIndex];
  if (segmentA.order !== segmentB.order) {
    return segmentA.order - segmentB.order;
  } 
  // Sort by reads' source track
  if (readA.sourceTrackID !== readB.sourceTrackID){
    return readA.sourceTrackID - readB.sourceTrackID;
  }
```